src/main/scala/FutureDeviceRep.scala

```diff
@@ -0,0 +1,27 @@
+import scala.concurrent.Future
+
+class FutureDeviceRep extends IotDeviceRepository[Future] {
+  override def registerDevice(userId: Long, serialNumber: String): Future[IotDevice] = {
+    val device = IotDevice( Storage.deviceStorageById.size, userId , serialNumber)
```

Comment:
this is not thread safe, if two threads call register device at the same time, you will end up with 2 devices having the same id.